File: app/src/main/java/com/fitbudget/app/data/seed/FoodSeed.kt
package com.fitbudget.app.data.seed

import com.fitbudget.app.data.database.entity.FoodEntity
import com.fitbudget.app.domain.model.FoodCategory
import com.fitbudget.app.domain.model.FoodRole

/**
 * Offline food database seed: common, low-cost Indian foods.
 *
 * Calories/protein are rounded reference values for a typical home-cooked serving and costs are
 * approximate Indian market prices. Everything here is an estimate and the user can edit or
 * add their own foods.
 */
object FoodSeed {

    fun foods(nowMillis: Long): List<FoodEntity> = raw.map { item ->
        FoodEntity(
            name = item.name,
            nameKey = item.name.trim().lowercase(),
            servingLabel = item.serving,
            calories = item.kcal,
            proteinG = item.protein,
            carbsG = item.carbs,
            fatG = item.fat,
            costRupees = item.cost,
            category = item.category,
            role = item.role,
            isCustom = false,
            createdAtMillis = nowMillis
        )
    }

    private data class Seed(
        val name: String,
        val serving: String,
        val kcal: Double,
        val protein: Double,
        val carbs: Double,
        val fat: Double,
        val cost: Double,
        val category: FoodCategory,
        val role: FoodRole
    )

    private val raw = listOf(
        // ---- Affordable protein ----
        Seed("Boiled Egg", "1 egg (50 g)", 78.0, 6.3, 0.6, 5.3, 7.0, FoodCategory.EGG, FoodRole.PROTEIN),
        Seed("Egg Omelette", "2 eggs + onion", 190.0, 13.0, 3.0, 14.0, 18.0, FoodCategory.EGG, FoodRole.PROTEIN),
        Seed("Egg White", "2 egg whites", 34.0, 7.2, 0.5, 0.1, 14.0, FoodCategory.EGG, FoodRole.PROTEIN),
        Seed("Toor Dal (cooked)", "1 bowl (150 g)", 150.0, 9.0, 21.0, 3.0, 12.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Masoor Dal (cooked)", "1 bowl (150 g)", 140.0, 9.0, 20.0, 2.0, 11.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Moong Dal (cooked)", "1 bowl (150 g)", 145.0, 9.5, 20.0, 2.0, 12.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Boiled Chana", "1 bowl (100 g)", 164.0, 8.9, 27.0, 2.6, 14.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Roasted Chana", "30 g", 110.0, 6.0, 18.0, 1.5, 5.0, FoodCategory.VEG, FoodRole.SNACK),
        Seed("Soya Chunks (cooked)", "50 g dry", 172.0, 26.0, 16.0, 0.5, 10.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Sprouted Moong", "1 bowl (100 g)", 120.0, 8.0, 19.0, 0.6, 10.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Rajma (cooked)", "1 bowl (150 g)", 180.0, 10.0, 30.0, 1.0, 16.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Paneer", "50 g", 145.0, 9.0, 2.0, 11.0, 25.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Peanuts (roasted)", "30 g", 170.0, 7.6, 6.0, 14.0, 6.0, FoodCategory.VEG, FoodRole.PROTEIN),
        Seed("Sattu Drink", "30 g sattu", 120.0, 6.0, 20.0, 1.5, 8.0, FoodCategory.VEG, FoodRole.BEVERAGE),
        Seed("Chicken Breast", "100 g", 165.0, 31.0, 0.0, 3.6, 40.0, FoodCategory.NON_VEG, FoodRole.PROTEIN),
        Seed("Chicken Curry", "1 bowl (150 g)", 240.0, 22.0, 6.0, 14.0, 60.0, FoodCategory.NON_VEG, FoodRole.PROTEIN),
        Seed("Rohu Fish (curry cut)", "100 g", 140.0, 20.0, 0.0, 6.0, 40.0, FoodCategory.NON_VEG, FoodRole.PROTEIN),

        // ---- Staples ----
        Seed("Roti (wheat)", "1 roti (40 g atta)", 110.0, 3.2, 22.0, 1.0, 3.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Rice (cooked)", "1 cup (150 g)", 200.0, 4.0, 44.0, 0.4, 6.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Poha", "1 plate (150 g)", 250.0, 5.0, 45.0, 6.0, 12.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Oats (cooked in water)", "40 g dry", 150.0, 5.0, 27.0, 3.0, 10.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Upma", "1 plate (200 g)", 230.0, 6.0, 38.0, 6.0, 12.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Idli", "2 pieces", 140.0, 4.0, 28.0, 0.6, 12.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Plain Dosa", "1 dosa", 160.0, 4.0, 26.0, 4.0, 15.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Whole Wheat Bread", "2 slices", 140.0, 5.0, 26.0, 2.0, 10.0, FoodCategory.VEG, FoodRole.STAPLE),
        Seed("Daliya (broken wheat)", "50 g dry", 170.0, 6.0, 34.0, 1.0, 8.0, FoodCategory.VEG, FoodRole.STAPLE),

        // ---- Dairy ----
        Seed("Toned Milk", "1 glass (200 ml)", 120.0, 6.4, 10.0, 5.0, 12.0, FoodCategory.VEG, FoodRole.DAIRY),
        Seed("Curd (dahi)", "1 bowl (150 g)", 90.0, 5.0, 7.0, 4.5, 12.0, FoodCategory.VEG, FoodRole.DAIRY),
        Seed("Buttermilk (chaas)", "1 glass (250 ml)", 40.0, 2.0, 4.0, 1.0, 8.0, FoodCategory.VEG, FoodRole.BEVERAGE),
        Seed("Ghee", "1 tsp (5 g)", 45.0, 0.0, 0.0, 5.0, 6.0, FoodCategory.VEG, FoodRole.SNACK),

        // ---- Vegetables ----
        Seed("Seasonal Mixed Vegetables", "1 bowl (150 g)", 90.0, 3.0, 14.0, 2.0, 12.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Boiled Potato", "1 medium (150 g)", 130.0, 3.0, 30.0, 0.2, 6.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Sweet Potato", "1 medium (150 g)", 130.0, 2.0, 30.0, 0.1, 12.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Tomato", "1 medium (100 g)", 22.0, 1.0, 4.0, 0.2, 4.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Onion", "1 medium (100 g)", 44.0, 1.2, 10.0, 0.1, 4.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Palak (spinach, cooked)", "1 bowl (150 g)", 45.0, 3.4, 5.0, 0.5, 8.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Cabbage Sabzi", "1 bowl (150 g)", 70.0, 2.0, 10.0, 2.5, 8.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Lauki (bottle gourd) Sabzi", "1 bowl (150 g)", 65.0, 1.5, 9.0, 2.5, 8.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Carrot", "1 medium (80 g)", 25.0, 0.6, 6.0, 0.1, 4.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Cucumber", "1 medium (150 g)", 30.0, 1.0, 5.0, 0.2, 5.0, FoodCategory.VEG, FoodRole.VEGETABLE),
        Seed("Green Salad", "1 plate", 50.0, 2.0, 8.0, 0.5, 10.0, FoodCategory.VEG, FoodRole.VEGETABLE),

        // ---- Fruit ----
        Seed("Banana", "1 medium (118 g)", 105.0, 1.3, 27.0, 0.4, 5.0, FoodCategory.VEG, FoodRole.FRUIT),
        Seed("Apple", "1 medium (180 g)", 95.0, 0.5, 25.0, 0.3, 18.0, FoodCategory.VEG, FoodRole.FRUIT),
        Seed("Guava", "1 medium (120 g)", 65.0, 2.6, 14.0, 0.9, 10.0, FoodCategory.VEG, FoodRole.FRUIT),
        Seed("Papaya", "1 bowl (150 g)", 60.0, 0.9, 15.0, 0.2, 12.0, FoodCategory.VEG, FoodRole.FRUIT),
        Seed("Orange", "1 medium (150 g)", 60.0, 1.2, 15.0, 0.2, 12.0, FoodCategory.VEG, FoodRole.FRUIT),
        Seed("Watermelon", "1 bowl (200 g)", 60.0, 1.2, 15.0, 0.3, 10.0, FoodCategory.VEG, FoodRole.FRUIT),

        // ---- Beverages / extras ----
        Seed("Green Tea", "1 cup", 2.0, 0.0, 0.0, 0.0, 4.0, FoodCategory.VEG, FoodRole.BEVERAGE),
        Seed("Tea with Milk", "1 cup (150 ml)", 90.0, 2.0, 12.0, 3.0, 8.0, FoodCategory.VEG, FoodRole.BEVERAGE),
        Seed("Black Coffee", "1 cup", 5.0, 0.3, 0.0, 0.0, 6.0, FoodCategory.VEG, FoodRole.BEVERAGE),
        Seed("Jaggery (gud)", "10 g", 38.0, 0.1, 9.5, 0.0, 2.0, FoodCategory.VEG, FoodRole.SNACK),
        Seed("Mustard Oil", "1 tsp (5 ml)", 45.0, 0.0, 0.0, 5.0, 2.0, FoodCategory.VEG, FoodRole.SNACK)
    )
}

File: ios/Sources/FitBudgetCore/Seed/FoodSeed.swift
import Foundation

/// Offline food database seed: common, low-cost Indian foods.
///
/// Calories/protein are rounded reference values for a typical home-cooked serving and costs are
/// approximate Indian market prices. Everything here is an estimate and the user can edit any food
/// or add their own.
public enum FoodSeed {

    public static let items: [FoodItem] = affordableProtein + staples + dairy + vegetables + fruit + extras

    // MARK: - Affordable protein

    private static let affordableProtein: [FoodItem] = [
        FoodItem(name: "Boiled Egg", servingLabel: "1 egg (50 g)", calories: 78, proteinG: 6.3,
                 carbsG: 0.6, fatG: 5.3, costRupees: 7, category: .egg, role: .protein),
        FoodItem(name: "Egg Omelette", servingLabel: "2 eggs + onion", calories: 190, proteinG: 13,
                 carbsG: 3, fatG: 14, costRupees: 18, category: .egg, role: .protein),
        FoodItem(name: "Egg White", servingLabel: "2 egg whites", calories: 34, proteinG: 7.2,
                 carbsG: 0.5, fatG: 0.1, costRupees: 14, category: .egg, role: .protein),
        FoodItem(name: "Toor Dal (cooked)", servingLabel: "1 bowl (150 g)", calories: 150, proteinG: 9,
                 carbsG: 21, fatG: 3, costRupees: 12, category: .veg, role: .protein),
        FoodItem(name: "Masoor Dal (cooked)", servingLabel: "1 bowl (150 g)", calories: 140, proteinG: 9,
                 carbsG: 20, fatG: 2, costRupees: 11, category: .veg, role: .protein),
        FoodItem(name: "Moong Dal (cooked)", servingLabel: "1 bowl (150 g)", calories: 145, proteinG: 9.5,
                 carbsG: 20, fatG: 2, costRupees: 12, category: .veg, role: .protein),
        FoodItem(name: "Boiled Chana", servingLabel: "1 bowl (100 g)", calories: 164, proteinG: 8.9,
                 carbsG: 27, fatG: 2.6, costRupees: 14, category: .veg, role: .protein),
        FoodItem(name: "Roasted Chana", servingLabel: "30 g", calories: 110, proteinG: 6,
                 carbsG: 18, fatG: 1.5, costRupees: 5, category: .veg, role: .snack),
        FoodItem(name: "Soya Chunks (cooked)", servingLabel: "50 g dry", calories: 172, proteinG: 26,
                 carbsG: 16, fatG: 0.5, costRupees: 10, category: .veg, role: .protein),
        FoodItem(name: "Sprouted Moong", servingLabel: "1 bowl (100 g)", calories: 120, proteinG: 8,
                 carbsG: 19, fatG: 0.6, costRupees: 10, category: .veg, role: .protein),
        FoodItem(name: "Rajma (cooked)", servingLabel: "1 bowl (150 g)", calories: 180, proteinG: 10,
                 carbsG: 30, fatG: 1, costRupees: 16, category: .veg, role: .protein),
        FoodItem(name: "Paneer", servingLabel: "50 g", calories: 145, proteinG: 9,
                 carbsG: 2, fatG: 11, costRupees: 25, category: .veg, role: .protein),
        FoodItem(name: "Peanuts (roasted)", servingLabel: "30 g", calories: 170, proteinG: 7.6,
                 carbsG: 6, fatG: 14, costRupees: 6, category: .veg, role: .protein),
        FoodItem(name: "Sattu Drink", servingLabel: "30 g sattu", calories: 120, proteinG: 6,
                 carbsG: 20, fatG: 1.5, costRupees: 8, category: .veg, role: .beverage),
        FoodItem(name: "Chicken Breast", servingLabel: "100 g", calories: 165, proteinG: 31,
                 carbsG: 0, fatG: 3.6, costRupees: 40, category: .nonVeg, role: .protein),
        FoodItem(name: "Chicken Curry", servingLabel: "1 bowl (150 g)", calories: 240, proteinG: 22,
                 carbsG: 6, fatG: 14, costRupees: 60, category: .nonVeg, role: .protein),
        FoodItem(name: "Rohu Fish (curry cut)", servingLabel: "100 g", calories: 140, proteinG: 20,
                 carbsG: 0, fatG: 6, costRupees: 40, category: .nonVeg, role: .protein)
    ]

    // MARK: - Staples

    private static let staples: [FoodItem] = [
        FoodItem(name: "Roti (wheat)", servingLabel: "1 roti (40 g atta)", calories: 110, proteinG: 3.2,
                 carbsG: 22, fatG: 1, costRupees: 3, category: .veg, role: .staple),
        FoodItem(name: "Rice (cooked)", servingLabel: "1 cup (150 g)", calories: 200, proteinG: 4,
                 carbsG: 44, fatG: 0.4, costRupees: 6, category: .veg, role: .staple),
        FoodItem(name: "Poha", servingLabel: "1 plate (150 g)", calories: 250, proteinG: 5,
                 carbsG: 45, fatG: 6, costRupees: 12, category: .veg, role: .staple),
        FoodItem(name: "Oats (cooked in water)", servingLabel: "40 g dry", calories: 150, proteinG: 5,
                 carbsG: 27, fatG: 3, costRupees: 10, category: .veg, role: .staple),
        FoodItem(name: "Upma", servingLabel: "1 plate (200 g)", calories: 230, proteinG: 6,
                 carbsG: 38, fatG: 6, costRupees: 12, category: .veg, role: .staple),
        FoodItem(name: "Idli", servingLabel: "2 pieces", calories: 140, proteinG: 4,
                 carbsG: 28, fatG: 0.6, costRupees: 12, category: .veg, role: .staple),
        FoodItem(name: "Plain Dosa", servingLabel: "1 dosa", calories: 160, proteinG: 4,
                 carbsG: 26, fatG: 4, costRupees: 15, category: .veg, role: .staple),
        FoodItem(name: "Whole Wheat Bread", servingLabel: "2 slices", calories: 140, proteinG: 5,
                 carbsG: 26, fatG: 2, costRupees: 10, category: .veg, role: .staple),
        FoodItem(name: "Daliya (broken wheat)", servingLabel: "50 g dry", calories: 170, proteinG: 6,
                 carbsG: 34, fatG: 1, costRupees: 8, category: .veg, role: .staple)
    ]

    // MARK: - Dairy

    private static let dairy: [FoodItem] = [
        FoodItem(name: "Toned Milk", servingLabel: "1 glass (200 ml)", calories: 120, proteinG: 6.4,
                 carbsG: 10, fatG: 5, costRupees: 12, category: .veg, role: .dairy),
        FoodItem(name: "Curd (dahi)", servingLabel: "1 bowl (150 g)", calories: 90, proteinG: 5,
                 carbsG: 7, fatG: 4.5, costRupees: 12, category: .veg, role: .dairy),
        FoodItem(name: "Buttermilk (chaas)", servingLabel: "1 glass (250 ml)", calories: 40, proteinG: 2,
                 carbsG: 4, fatG: 1, costRupees: 8, category: .veg, role: .beverage),
        FoodItem(name: "Ghee", servingLabel: "1 tsp (5 g)", calories: 45, proteinG: 0,
                 carbsG: 0, fatG: 5, costRupees: 6, category: .veg, role: .snack)
    ]

    // MARK: - Vegetables

    private static let vegetables: [FoodItem] = [
        FoodItem(name: "Seasonal Mixed Vegetables", servingLabel: "1 bowl (150 g)", calories: 90,
                 proteinG: 3, carbsG: 14, fatG: 2, costRupees: 12, category: .veg, role: .vegetable),
        FoodItem(name: "Boiled Potato", servingLabel: "1 medium (150 g)", calories: 130, proteinG: 3,
                 carbsG: 30, fatG: 0.2, costRupees: 6, category: .veg, role: .vegetable),
        FoodItem(name: "Sweet Potato", servingLabel: "1 medium (150 g)", calories: 130, proteinG: 2,
                 carbsG: 30, fatG: 0.1, costRupees: 12, category: .veg, role: .vegetable),
        FoodItem(name: "Tomato", servingLabel: "1 medium (100 g)", calories: 22, proteinG: 1,
                 carbsG: 4, fatG: 0.2, costRupees: 4, category: .veg, role: .vegetable),
        FoodItem(name: "Onion", servingLabel: "1 medium (100 g)", calories: 44, proteinG: 1.2,
                 carbsG: 10, fatG: 0.1, costRupees: 4, category: .veg, role: .vegetable),
        FoodItem(name: "Palak (spinach, cooked)", servingLabel: "1 bowl (150 g)", calories: 45,
                 proteinG: 3.4, carbsG: 5, fatG: 0.5, costRupees: 8, category: .veg, role: .vegetable),
        FoodItem(name: "Cabbage Sabzi", servingLabel: "1 bowl (150 g)", calories: 70, proteinG: 2,
                 carbsG: 10, fatG: 2.5, costRupees: 8, category: .veg, role: .vegetable),
        FoodItem(name: "Lauki (bottle gourd) Sabzi", servingLabel: "1 bowl (150 g)", calories: 65,
                 proteinG: 1.5, carbsG: 9, fatG: 2.5, costRupees: 8, category: .veg, role: .vegetable),
        FoodItem(name: "Carrot", servingLabel: "1 medium (80 g)", calories: 25, proteinG: 0.6,
                 carbsG: 6, fatG: 0.1, costRupees: 4, category: .veg, role: .vegetable),
        FoodItem(name: "Cucumber", servingLabel: "1 medium (150 g)", calories: 30, proteinG: 1,
                 carbsG: 5, fatG: 0.2, costRupees: 5, category: .veg, role: .vegetable),
        FoodItem(name: "Green Salad", servingLabel: "1 plate", calories: 50, proteinG: 2,
                 carbsG: 8, fatG: 0.5, costRupees: 10, category: .veg, role: .vegetable)
    ]

    // MARK: - Fruit

    private static let fruit: [FoodItem] = [
        FoodItem(name: "Banana", servingLabel: "1 medium (118 g)", calories: 105, proteinG: 1.3,
                 carbsG: 27, fatG: 0.4, costRupees: 5, category: .veg, role: .fruit),
        FoodItem(name: "Apple", servingLabel: "1 medium (180 g)", calories: 95, proteinG: 0.5,
                 carbsG: 25, fatG: 0.3, costRupees: 18, category: .veg, role: .fruit),
        FoodItem(name: "Guava", servingLabel: "1 medium (120 g)", calories: 65, proteinG: 2.6,
                 carbsG: 14, fatG: 0.9, costRupees: 10, category: .veg, role: .fruit),
        FoodItem(name: "Papaya", servingLabel: "1 bowl (150 g)", calories: 60, proteinG: 0.9,
                 carbsG: 15, fatG: 0.2, costRupees: 12, category: .veg, role: .fruit),
        FoodItem(name: "Orange", servingLabel: "1 medium (150 g)", calories: 60, proteinG: 1.2,
                 carbsG: 15, fatG: 0.2, costRupees: 12, category: .veg, role: .fruit),
        FoodItem(name: "Watermelon", servingLabel: "1 bowl (200 g)", calories: 60, proteinG: 1.2,
                 carbsG: 15, fatG: 0.3, costRupees: 10, category: .veg, role: .fruit)
    ]

    // MARK: - Beverages and extras

    private static let extras: [FoodItem] = [
        FoodItem(name: "Green Tea", servingLabel: "1 cup", calories: 2, proteinG: 0,
                 carbsG: 0, fatG: 0, costRupees: 4, category: .veg, role: .beverage),
        FoodItem(name: "Tea with Milk", servingLabel: "1 cup (150 ml)", calories: 90, proteinG: 2,
                 carbsG: 12, fatG: 3, costRupees: 8, category: .veg, role: .beverage),
        FoodItem(name: "Black Coffee", servingLabel: "1 cup", calories: 5, proteinG: 0.3,
                 carbsG: 0, fatG: 0, costRupees: 6, category: .veg, role: .beverage),
        FoodItem(name: "Jaggery (gud)", servingLabel: "10 g", calories: 38, proteinG: 0.1,
                 carbsG: 9.5, fatG: 0, costRupees: 2, category: .veg, role: .snack),
        FoodItem(name: "Mustard Oil", servingLabel: "1 tsp (5 ml)", calories: 45, proteinG: 0,
                 carbsG: 0, fatG: 5, costRupees: 2, category: .veg, role: .snack)
    ]
}
